### PR TITLE
Inspect: Add error tab

### LINKED ISSUE #21482: PanelInspect: Add error tab
Add a error tab to Panel inspector and render the error in tabcontent

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds an error tab to the inspect drawer where you can view panel errors. Also makes tabs conditional in the sense that we only show them if they have some information.

**Which issue(s) this PR fixes**:
Fixes #21482

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mckn — APPROVED** [review]
Have not tested it but the code looks really good! Great work!

File: public/app/features/dashboard/components/Inspector/PanelInspector.tsx [hunk 1]
```diff
@@ -28,6 +30,9 @@ interface State {
   // Data frem the last response
   data: DataFrame[];
 
+  // Error from query
+  error: DataQueryError;
+
```
Comment (ryantxu) [inline public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: do we need a special entry for error here?  or just check `last.error`?  Eventually the `last` variable should be better typed....

File: public/app/features/dashboard/components/Inspector/PanelInspector.tsx [hunk 2]
```diff
@@ -82,6 +88,8 @@ export class PanelInspector extends PureComponent<Props, State> {
     // Find the first DataSource wanting to show custom metadata
     let metaDS: DataSourceApi;
     const data = lastResult?.series as DataFrame[];
+    const error = lastResult?.error as DataQueryError;
```
Comment (dprokop) [inline public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: Can we type `lastResult` as `PanelData` to remove casting?

File: public/app/features/dashboard/components/Inspector/PanelInspector.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: Agree with @ryantxu, we probably don't need this.

File: public/app/features/dashboard/components/Inspector/PanelInspector.tsx [hunk 3]
```diff
@@ -73,15 +83,17 @@ export class PanelInspector extends PureComponent<Props, State> {
     }
 
     // TODO? should we get the result with an observable once?
-    const lastResult = (panel.getQueryRunner() as any).lastResult;
+    const lastResult: PanelData = (panel.getQueryRunner() as any).lastResult;
```
Comment (dprokop) [inline public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: @torkelo what's the suggested way here? Accessing private prop feels wrong

Comment (torkelo) [reply public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: yea, maybe add a function getLastResult()

Comment (dprokop) [reply public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: Yup, this feels better, but I was thinking about observables way. However, maybe that's too simple to use observables.

**Review by dprokop — COMMENTED** [review]
Code wise it looks good. I have noticed couple of UI things worth fixing before merging this:
- [ ] The scroll behaviour of the drawer is weird, as it scrolls the entire content container. This results in content overlapping the panels title bar. Also, the tabs itself are scrolled:
image

Think the tab's pane should be scrollable only in Panel inspect.

- [ ] Not sure how to best approach this to improve the error readability, but maybe we could at least render it as `<code>` / `<pre>`

**Comment by peterholmberg** [comment]
@dprokop this is _very_ weird as this behaviour works in storybook. I am investigating this since it is not very useable at the moment. But still weird.

File: packages/grafana-ui/src/components/Drawer/Drawer.tsx [hunk 4]
```diff
@@ -53,8 +54,10 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => {
       justify-content: center;
     `,
     content: css`
-      padding: ${theme.spacing.md};
-      height: 100%;
+      padding: ${theme.spacing.sm} ${theme.spacing.xs};
+      height: 95%;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Drawer/Drawer.tsx]: Why 95%?

File: packages/grafana-ui/src/components/Tabs/TabContent.tsx [hunk 5]
```diff
@@ -10,9 +10,8 @@ interface Props {
 const getTabContentStyle = stylesFactory((theme: GrafanaTheme) => {
   return {
     tabContent: css`
-      padding: ${theme.spacing.xs};
-      height: 90%;
-      overflow: hidden;
+      padding: ${theme.spacing.sm};
+      height: 95%;
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Tabs/TabContent.tsx]: I don't think TabContent should have this set

File: public/app/features/dashboard/components/Inspector/PanelInspector.tsx [hunk 6]
```diff
@@ -204,9 +208,11 @@ export class PanelInspector extends PureComponent<Props, State> {
     }
     if (error.data) {
       return (
-        <div>
+        <div style={{ overflowY: 'auto', height: '100%' }}>
```
Comment (dprokop) [inline public/app/features/dashboard/components/Inspector/PanelInspector.tsx]: Uhm, what about wrapping every but Table content with CustomScrollbar for consistency?

**Comment by dprokop** [comment]
image
Raw JSON pane does not scroll

File: packages/grafana-ui/src/components/Drawer/Drawer.tsx (hunk 4 above)
Comment (peterholmberg) [reply packages/grafana-ui/src/components/Drawer/Drawer.tsx]: Did you try setting this to auto?

Comment (dprokop) [reply packages/grafana-ui/src/components/Drawer/Drawer.tsx]: nope

**Review by dprokop — APPROVED** [review]
I think we should merge this now and wait with improvements where the design is in place. The scrolling works fine now with a small note about inconsistency in the error pan -> `pre` tag has `overflow: auto` (global styles) making it scrollable. This means that in the error tab the Custom scrollbar never kicks in. But let's wait for the design.

**Pull request merged**